### PR TITLE
Make it more obvious we are building a JSON API

### PR DESCRIPTION
**Pull request by paulbrimicombe**
## What?

Remove auto-deserialisation of JSON API responses to Java classes in integration tests

## Why?

Having been involved in quite a few of these coding exercises, candidates often don't realise / engage with the fact that they are building a RESTful JSON API and often don't realise that the consuming code may well not be Java.

The way the integration tests immediately and automatically deserialise the API responses into a `Scoreboard` object means that candidates don't don't consider the shape of the JSON response and whether all of the required information is available in the response without further processing client-side.

Removing this automatic deserialisation also means that we don't need the "magical" annotations (see also #42) which can be confusing / unfamiliar to candidates.

## Open questions / comments

- With this PR, the testing would require use of the Jackson API to extract data from the `ObjectNode` (e.g. `scoreboard.get("totals").get("LAB").asInt()` to extract the total from `{ "totals": { "LAB": 123 } }`). Should we give candidates some hints about how to extract this data in case they are unfamiliar with the API? Or is it OK to expect them to be able to work that out as they go along?
- Sometimes having a non-ideal solution can help candidates because they can identify improvements / changes / have something to talk about. Could this be one of those situations?
- Is it only me that has experienced this problem??

### CODE REVIEW

**Comment by jonpbbc** [comment]
> Should we give candidates some hints about how to extract this data in case they are unfamiliar with the API? Or is it OK to expect them to be able to work that out as they go along?

My gut feeling is any candidates that don't have an IDE with auto-completion based on object types configured are likely to get hung up on this access pattern, particularly more junior ones, and that it could prevent us from seeing how they approach the logical problem As we're not testing their knowledge of the specific API it feels reasonable to give an example of how to access the data (maybe something like `int labVotes = scoreboard.get("totals").get("LAB").asInt();` in the first test body?) to mitigate that.

File: election-api-java/src/test/java/bbc/news/elections/ElectionsApiApplicationIntegrationTests.java [hunk 1]
```diff
@@ -44,20 +48,42 @@ public void reset() throws Exception {
...
 	public void first5Test() throws Exception {
-		Scoreboard scoreboard = runTest(5);
+		JsonNode scoreboard = runTest(5);
 
 		assertNotNull(scoreboard);
 		// assert LD == 1
 		// assert LAB = 4
 		// assert winner = noone
+
+		// Some Jackson JSON API examples:
```
Comment (paulbrimicombe) [inline election-api-java/src/test/java/bbc/news/elections/ElectionsApiApplicationIntegrationTests.java:59]: This set of examples takes up a fair bit of space but I didn't want to steer the candidates towards a specific data structure (e.g. key-value object vs. array).

**Comment by paulbrimicombe** [comment]
> My gut feeling is any candidates that don't have an IDE with auto-completion based on object types configured are likely to get hung up on this access pattern, particularly more junior ones, and that it could prevent us from seeing how they approach the logical problem As we're not testing their knowledge of the specific API it feels reasonable to give an example of how to access the data (maybe something like `int labVotes = scoreboard.get("totals").get("LAB").asInt();` in the first test body?) to mitigate that.

Thanks for the comment @jonpbbc — I've added some examples, trying not to guide candidates towards a specific shape of response.

**Comment by jonpbbc** [comment]
Looks good to me; probably needs a second reviewer if possible.

**Pull request closed without merging**